theurgy.lic

```diff
@@ -2,185 +2,453 @@
+    @safe_room = @settings.safe_room
+
+    @rituals = get_rituals
+    @communes = get_communes
+
+    annotate_item_presence(@rituals + @communes)
+    
+    @rituals.reject! do |r|
+      r[:items].any? { |i| !i[:on_hand] && !i[:shop] }
+    end
+    @communes.reject! do |c|
+      c[:items].any? { |i| !i[:on_hand] && !i[:shop] }
+    end
+  end
 
-  ######################## GET/PUT ########################
-
-  def put_in_container(item)
-    bput "put #{item} in my #{@theurgy_supply_container}", 'You put', 'What were you referring to'
+  def get_rituals
+    rituals = all_rituals
+    blacklist = @settings.theurgy_blacklist
+    rituals.reject! { |r| blacklist.include? r[:method].to_s }
+    whitelist = @settings.theurgy_whitelist
+    unless whitelist.empty?
+      rituals.select! { |r| whitelist.include? r[:method].to_s }
+    end
+    rituals
+  end
+  
+  def all_rituals
+    rituals = []
+    if @data['almsbox'] && @settings.tithe
+      rituals << {
+        method: :tithe,
+      }
+    end
+    if @hometown == 'Crossing'
+      if DRStats.circle >= 30
+        rituals << {
+          method: :study_wall,
+        }
+      end
+      rituals << {
+        method: :refectory,
+      }
+    end
+    if DRSkill.getrank('Mechanical Lore') > 140 && @immortal_aspect && @data['limb_foraging']
+      # TODO: allow other knife nouns 
+      rituals << {
+        method: :carve_bead,
+        items: [
+          { name: 'carving knife', any_container: true },
+          { name: 'shaper', any_container: true},
+          { name: 'prayer chain', any_container: true },
+        ],
+      }
+    end
+    if @data['gather_sirese'] && @data['plant_sirese']
+      rituals << {
+        method: :sirese_seed,
+      }
+    end
+    if @data['bath']
+      herbs = @data['herbs'] || ['sage', 'lavender']
+      rituals << {
+        method: :bathe,
+        items: herbs.map { |h| { name: h, shop: @data['theurgy_supplies'] } },
+      }
+    end
+    if @immortal_aspect
+      ritals << {
+        method: :meditate_bead,
+        items: [
+          { name: 'prayer chain', any_container: true },
+        ],
+      }
+    end
+    rituals << {
+      method: :pray_badge,
+    }
+    # TODO: prayer mat support
+    if @data['altar']
+      if DRStats.circle >= 10
+        rituals << {
+          method: :dance,
+        }
+      end
+      rituals << {
+        method: :incense,
+        items: [
+          { name: 'incense', shop: @data['theurgy_supplies'] },
+          { name: 'flint', shop: @data['flint_shop'] },
+        ]
+      }
+      rituals << {
+        method: :wine,
+        items: [
+          { name: 'wine', shop: @data['theurgy_supplies'] },
+        ],
+      }
+      rituals << {
+        method: :recite_prayer,
+      }
+      rituals << {
+        method: :clean_altar,
+      }
+    end
+    rituals << {
+      method: :embarass_myself,
+      items: [
+        {
+          name: 'golden parchment',
```

Comment:
See about 'customization for the sake of customization'

Reply:
This isn't for the sake of customization. This is a necessity because the same prayer scroll is obnoxious and I've complained about it before. If this is the case, then why do we support different types of lockpicks? Everyone should get a standard one. Personally, I think this entire reciting of the parchment is obnoxious on its own, but giving options doesn't hurt anything.

Reply:
Customization makes the code more robust, and helps alleviate some of the gripes people have with it.

Reply:
There's a reason I added this ritual to the blacklist by default -- I also think it's obnoxious.

I'd like to punt this particular request to the future, though, as I suspect a lot of people don't actually want to do this ritual, and won't do it once it's turned off by default.

Reply:
It is super useful on a brand new cleric though, even if its not at 150

Reply:
I'm with you, I removed it as it wasn't needed for capping Theurgy.